File: android/src/main/kotlin/com/example/simple_widget/ImageFileManager.kt
package com.example.simple_widget

import android.content.Context
import java.io.File
import java.util.UUID

object ImageFileManager {

  private const val IMAGE_DIR = "widget_images"

  fun getImageDir(context: Context): File {
    val dir = File(context.filesDir, IMAGE_DIR)
    if (!dir.exists()) {
      dir.mkdirs()
    }
    return dir
  }

  fun saveImage(context: Context, bytes: ByteArray, filename: String?): String {
    val dir = getImageDir(context)
    val name = filename ?: "${UUID.randomUUID()}.png"
    val file = File(dir, name)
    file.writeBytes(bytes)
    return "$IMAGE_DIR/$name"
  }

  fun deleteImages(context: Context, relativePaths: List<String>) {
    for (path in relativePaths) {
      val file = File(context.filesDir, path)
      if (file.exists()) {
        file.delete()
      }
    }
  }

  fun readImage(context: Context, relativePath: String): ByteArray? {
    val file = File(context.filesDir, relativePath)
    if (file.exists()) {
      return file.readBytes()
    }
    return null
  }

  fun getBasePath(context: Context): String {
    return context.filesDir.absolutePath
  }
}

File: ios/Classes/ImageFileManager.swift
//
//  ImageFileManager.swift
//  simple_widget
//

import Foundation

class ImageFileManager {

  static let shared = ImageFileManager()

  private let imageDir = "widget_images"

  func getImageDirectory() -> URL? {
    guard let groupId = Settings.groupId else {
      return nil
    }
    guard let containerURL = FileManager.default.containerURL(forSecurityApplicationGroupIdentifier: groupId) else {
      return nil
    }
    let dir = containerURL.appendingPathComponent(imageDir)
    if !FileManager.default.fileExists(atPath: dir.path) {
      try? FileManager.default.createDirectory(at: dir, withIntermediateDirectories: true)
    }
    return dir
  }

  func saveImage(_ bytes: Data, filename: String?) -> String? {
    guard let dir = getImageDirectory() else { return nil }
    let name = filename ?? "\(UUID().uuidString).png"
    let file = dir.appendingPathComponent(name)
    do {
      try bytes.write(to: file)
      return "\(imageDir)/\(name)"
    } catch {
      print("Failed to save image: \(error)")
      return nil
    }
  }

  func deleteImages(_ relativePaths: [String]) {
    guard let groupId = Settings.groupId,
          let containerURL = FileManager.default.containerURL(forSecurityApplicationGroupIdentifier: groupId) else {
      return
    }
    for path in relativePaths {
      let file = containerURL.appendingPathComponent(path)
      try? FileManager.default.removeItem(at: file)
    }
  }

  func getBasePath() -> String? {
    guard let groupId = Settings.groupId,
          let containerURL = FileManager.default.containerURL(forSecurityApplicationGroupIdentifier: groupId) else {
      return nil
    }
    return containerURL.path
  }
}
